src/sorted_container_iteration.jl

```diff
@@ -0,0 +1,1167 @@
+end
+
+"""
+    Base.first(sc::SortedContainer)
+
+Return the 
+first item (a `k=>v` pair for SortedDict and
+SortedMultiDict or an element for SortedSet) in `sc` according to the sorted
+order in the container. It is a `BoundsError` to call this function on
+an empty container.  Equivalent to `deref(token_startindex(sc))`. Time: O(log *n*)
+"""
+Base.first(m::SortedContainer) = deref(token_firstindex(m))
+
+
+"""
+    Base.last(sc::SortedContainer)
+
+Return the last item (a `k=>v` pair for SortedDict and
+SortedMultiDict or a key for SortedSet) in `sc` according to the sorted
+order in the container. It is a `BoundsError` to call this function on an
+empty container. Equivalent to `deref(token_lastindex(sc))`. Time: O(log *n*)
+"""
+Base.last(m::SortedContainer) = deref(token_lastindex(m))
+
+
+
+"""
+    Base.getindex(m::SortedDict, st::IntSemiToken)
+    Base.getindex(m::SortedMultiDict, st::IntSemiToken)
+
+Retrieve value portion of item from SortedDict or SortedMultiDict
+`m` indexed by `st`, a semitoken. Notation `m[st]` appearing in 
+an expression
+is equivalent to [`deref_value(token::Token)`](@ref) where `token=(m,st)`.  Time: O(1)
+"""
+function Base.getindex(m::SortedDict,
+                       i::IntSemiToken)
+    @boundscheck has_data((m,i))
+    @inbounds d = m.bt.data[i.address].d
+    return d
+end
+# Must repeat this to break ambiguity; cannot use SDMContainer.
+function Base.getindex(m::SortedMultiDict,
+                       i::IntSemiToken)
+    @boundscheck has_data((m,i))
+    @inbounds d = m.bt.data[i.address].d
+    return d
+end
+
+
+
+"""
+    Base.setindex!(m::SortedDict, newvalue, st::IntSemiToken)
+    Base.setindex!(m::SortedMultiDict, newvalue, st::IntSemiToken)
+
+Set the value portion of item from SortedDict or SortedMultiDict
+`m` indexed by `st`, a semitoken to `newvalue`.  Time: O(1)
+"""
+function Base.setindex!(m::SortedDict,
+                        d_,
+                        i::IntSemiToken)
+    @boundscheck has_data((m,i))
+    @inbounds m.bt.data[i.address] =
+        KDRec{keytype(m),valtype(m)}(m.bt.data[i.address].parent,
+                                     m.bt.data[i.address].k,
+                                     convert(valtype(m),d_))
+    return m
+end
+## Must repeat this to break ambiguity; cannot use SDMContainer
+function Base.setindex!(m::SortedMultiDict,
+                        d_,
+                        i::IntSemiToken)
+    @boundscheck has_data((m,i))
+    @inbounds m.bt.data[i.address] =
+        KDRec{keytype(m),valtype(m)}(m.bt.data[i.address].parent,
+                                     m.bt.data[i.address].k,
+                                     convert(valtype(m),d_))
+    return m
+end
+
+
+"""
+    Base.searchsortedfirst(m::SortedContainer, k)
+
+Return the semitoken of the first item in the 
+sorted container `m` that is greater than or equal to
+`k` in the sort order.
+If there is no
+such item, then the past-end semitoken is returned.  Time: O(*c* log *n*)
```

Comment:
I didn't know this!  However, if you don't mind, I'll leave it as it is because there is not much benefit in changing it compared to the effort needed.